storage/redis_adapter.go

```diff
@@ -3,13 +3,31 @@ package storage
 import (
 	"fmt"
 	"log"
+	"time"
 
 	r "gopkg.in/redis.v3"
 )
 
+type message struct {
+	app         string
+	messageBody string
+}
+
+func newMessage(app string, messageBody string) *message {
+	return &message{
+		app:         app,
+		messageBody: messageBody,
+	}
+}
+
 type redisAdapter struct {
-	bufferSize  int
-	redisClient *r.Client
+	bufferSize     int
+	redisClient    *r.Client
+	messageCount   int
+	messageChannel chan *message
+	pipeline       *r.Pipeline
+	timeoutTicker  *time.Ticker
+	queuedApps     map[string]bool
```

Comment:
the only argument I have is concurrency safety. it looks like the only place that most of these are accessed is inside of the `start()` func, but it would be much easier to follow and prove concurrency safety if these values were explicitly passed across goroutine boundaries, or from function to function inside a single goroutine